providers/common/kdfs/sskdf.c

```diff
@@ -288,172 +301,70 @@ static int SSKDF_mac_kdm(EVP_MAC *kdf_mac, const EVP_MD *hmac_md,
     return ret;
 }
 
-static EVP_KDF_IMPL *sskdf_new(void)
+static void *sskdf_new(void *provctx)
 {
-    EVP_KDF_IMPL *impl;
+    KDF_SSKDF *ctx;
 
-    if ((impl = OPENSSL_zalloc(sizeof(*impl))) == NULL)
-        KDFerr(KDF_F_SSKDF_NEW, ERR_R_MALLOC_FAILURE);
-    return impl;
+    if ((ctx = OPENSSL_zalloc(sizeof(*ctx))) == NULL)
+        ERR_raise(ERR_LIB_PROV, ERR_R_MALLOC_FAILURE);
+    ctx->provctx = provctx;
+    return ctx;
 }
 
-static void sskdf_reset(EVP_KDF_IMPL *impl)
+static void sskdf_reset(void *vctx)
 {
-    OPENSSL_clear_free(impl->secret, impl->secret_len);
-    OPENSSL_clear_free(impl->info, impl->info_len);
-    OPENSSL_clear_free(impl->salt, impl->salt_len);
-    EVP_MAC_free(impl->mac);
+    KDF_SSKDF *ctx = (KDF_SSKDF *)vctx;
+
+    OPENSSL_clear_free(ctx->secret, ctx->secret_len);
+    OPENSSL_clear_free(ctx->info, ctx->info_len);
+    OPENSSL_clear_free(ctx->salt, ctx->salt_len);
+    EVP_MAC_free(ctx->mac);
 #if 0                    /* TODO(3.0) When we switch to fetched MDs */
```

Comment:
Not sure what this means..

Reply:
Me either.

Reply:
@levitte ?

Reply:
This is from before MDs in providers were available and sane (remember that they were lacking all sorts of info at one time, making direct fetches unusable?)...  so this kind of `#if 0` should be safe to remove now.

Reply:
This one is gone now.